File: project/inung-onboarding/src/main/java/icd/onboarding/surveyproject/survey/repository/SurveyRepository.java
package icd.onboarding.surveyproject.survey.repository;

import icd.onboarding.surveyproject.survey.service.domain.Survey;

import java.util.Optional;
import java.util.UUID;

public interface SurveyRepository {
	Survey save (Survey survey);

	Optional<Survey> findByIdAndVersion (UUID id, int version);
}

File: project/kyungrok-onboarding/src/main/kotlin/kr/innercircle/onboarding/survey/repository/SurveyRepository.kt
package kr.innercircle.onboarding.survey.repository

import kr.innercircle.onboarding.survey.domain.Survey

/**
 * packageName : kr.innercircle.onboarding.survey.repository
 * fileName    : SurveyRepository
 * author      : ckr
 * date        : 25. 6. 16.
 * description :
 */
interface SurveyRepository {
    fun save(survey: Survey): Survey
    fun findById(id: Long): Survey?
    fun findAll(): List<Survey>
}